### PR TITLE
faster few span methods

### PR DESCRIPTION
**Pull request by klensy**
Touched few methods, so it should be (hopefully) faster.

First two changes: instead splitting string from start and taking only last piece, split it from the end.
Last: swapped conditions, to first check boolean parameter.

### CODE REVIEW

File: compiler/rustc_span/src/source_map.rs [hunk 1]
```diff
@@ -632,10 +632,11 @@ impl SourceMap {
...
         match self.span_to_prev_source(sp) {
             Err(_) => None,
-            Ok(source) => source
-                .split('\n')
-                .last()
-                .map(|last_line| last_line.len() - last_line.trim_start().len()),
+            Ok(source) => {
+                let last_line = source.rsplit_once('\n').unwrap_or(("", &source)).1;
+
+                Some(last_line.len() - last_line.trim_start().len())
```
Comment (varkor) [inline compiler/rustc_span/src/source_map.rs:638]: This now returns `Some` instead of `None` when there is no `\n`. Was this an intentional change?

Comment (klensy) [reply compiler/rustc_span/src/source_map.rs:638]: If there any string, even without `\n`, it won't return `None`:
```rust
assert_eq!(Some("abcdef"), "abcdef".split('\n').last());
```

Comment (varkor) [reply compiler/rustc_span/src/source_map.rs:638]: Ah, of course, apologies.

**Comment by varkor** [comment]
Thanks for the pull request!

@bors r+ rollup

**Pull request merged**